MobiusCore/Source/Mobius.swift

```diff
@@ -19,15 +19,9 @@
 
 import Foundation
 
-public protocol LoopTypes {
-    associatedtype Model
-    associatedtype Event
-    associatedtype Effect
-}
-
-public typealias Update<T: LoopTypes> = (T.Model, T.Event) -> Next<T.Model, T.Effect>
+public typealias Update<Model, Event, Effect> = (Model, Event) -> Next<Model, Effect>
 
-public typealias Initiator<T: LoopTypes> = (T.Model) -> First<T.Model, T.Effect>
+public typealias Initiator<Model, Effect> = (Model) -> First<Model, Effect>
```

Comment:
Turns out the initiator isn't using `Event` even though it is generic over it. Such phantom types can hurt type inference.